Sources/MapboxCoreNavigation/HistoryEvent.swift

```diff
@@ -49,14 +49,17 @@ internal class StatusUpdateHistoryEvent: UnknownHistoryEvent {
     }
 }
 
-internal class PushRecordHistoryEvent: UnknownHistoryEvent {
-    let type: String
-    let properties: String
+/// History event being pushed by the user
+public struct PushRecordHistoryEvent: HistoryEvent {
+    public let timestamp: TimeInterval
+    /// The event type in the events log for this custom event.
+    public let type: String
+    /// The data value that contains a valid JSON attached to the event.
```

Comment:
'valid JSON' means its syntactic validity. How could I emphasize it more? There are no requirements to the contents of this data since it's filled by the user.